### ISSUE TITLE
Conflicting openai dependencies

### ISSUE THREAD

**Issue by mwildehahn** [stanfordnlp/dspy#583]
There are two dependency declarations for openai:
- https://github.com/stanfordnlp/dspy/blob/c2d639fb1c1dafa7d6a12b3fe730b3564580fbfe/pyproject.toml#L79
- https://github.com/stanfordnlp/dspy/blob/c2d639fb1c1dafa7d6a12b3fe730b3564580fbfe/pyproject.toml#L25

I'm trying to install this into my poetry managed app and it's complaining about the one pinned to 0.28.1.

It seems like the answer to this: https://github.com/stanfordnlp/dspy/blob/c2d639fb1c1dafa7d6a12b3fe730b3564580fbfe/pyproject.toml#L21 is yes? Happy to contribute moving off the old dependencies array to poetry if that is desired.

**Comment by arnavsinghvi11** [comment]
@mwildehahn agreed that including both dependencies can cause internal conflicts. would be great if you could help with the dependencies migration!
tagging @okhat  as well

**Comment by darinkishore** [comment]
updated the poetry dependency. 

but i think keeping both dependencies is best -> tools.poetry.dependencies is used for poetry users to be able to just `poetry install` things. 

however, the project dependencies section functions similarly to `requirements.txt` ->

see: https://stackoverflow.com/questions/62408719/download-dependencies-declared-in-pyproject-toml-using-pip

tldr: `python -m pip install .` will install dependencies defined in `pyproject.toml` under the first `dependencies` list
